tests/runtime/test_dlopen.c

```diff
@@ -43,13 +43,15 @@ main (int argc, char **argv)
               SHLIB_ORIGIN "/../../lib/CL/devices/%s/libpocl-devices-%s.so", argv[1],
               argv[1]);
 
+#ifdef ENABLE_ICD
```

Comment:
This causes the breakage. The file should include config.h and use BUILD_ICD macro instead. Now it's never loading the libpocl.so, thus its utility functions are not imported unless linking statically. If I fix that, vulkan works, but then I see some other failures with LLVM 10 and conformance test micro.

Reply:
(Managed to successfully run my test envs, pushing after another clean run)